### DISCUSSION TITLE
`mill init` shouldn't warn about `.mill-version` or leave behind an `out`.

### DISCUSSION THREAD

**Discussion by ckipp01** [com-lihaoyi/mill#1985] [Ideas]
I recently tried out the new `mill init` command when making https://github.com/ckipp01/mill-plugin.g8, but there were a couple things I noticed that I think may surprise users and are undesirable.

When you attempt to create a repo you'll almost always see:
```
❯ mill init ckipp01/mill-plugin.g8
No mill version specified.
You should provide a version via '.mill-version' file or --mill-version option.
Retrieving latest mill version ...
Using mill version 0.10.5
No build file found, you should create build.sc to do something useful
[1/1] init > [1/1] mill.scalalib.giter8.Giter8Module.init
Creating a new project...
```

1. I think the vast majority of the time when someone is creating a new repo from a template they're not going to be inside of an existing Mill build. Having a message that tells the user they need to have a `.mill-version` or that they should create a build file because none is found is quite confusing because they don't want those things where they're at, they just want to create their new repo.
2. After a correct creation of your new repo in your current repo there is now a stray `out` that is left where the user doesn't want it. `sbt` use to have this issue with `sbt new ` as well with leaving behind a `target`, but that's no longer the case and it's much nicer and cleaner to not have to cleanup a stray directory.

Are either of these addressable?

**Comment by lefou** [comment]
You are addressing three different issues here:

1. No Mill version specified

> ```
> No mill version specified.
> You should provide a version via '.mill-version' file or --mill-version option.
> Retrieving latest mill version ...
> Using mill version 0.10.5

Those lines come from https://github.com/lefou/millw and can not be fixed in this repo.

2. No Mill Buildfile

> ```
> No build file found, you should create build.sc to do something useful
> ```

This could be fixed by simple removing this output. But it was added in the past, to avoid other confusion, so we should probably only hide this message when Mill is returning successfully, but keep it in case we run into errors, so the user has a first hint about the cause (e.g. wrong directory).

3. Unwanted `out` directory creation

This one is harder to solve. I currently cannot find the right issue, but I think it was discussed before. I agree, we should only create the `out` dir once we start the Mill evaluator. But currently we also place the Mill server process socket there, which we probably should instead create in some `XDG` directory.

**Comment by ckipp01** [comment]
> Those lines come from https://github.com/lefou/millw and can not be fixed in this repo.

Ahhh sorry! I always forget that I have millw as my global and the mainline mill script in the local repo 😅 .

> This could be fixed by simple removing this output. But it was added in the past, to avoid other confusion, so we should probably only hide this message when Mill is returning successfully, but keep it in case we run into errors, so the user has a first hint about the cause (e.g. wrong directory).

Yea this seems totally reasonable.
